pimd/pim6_cmd.c

```diff
@@ -33,16 +33,445 @@
+
+	if (detail)
+		pim_show_interfaces_single(vrf_struct->info, vty, detail,
+					   mlag_bool, json_parent);
+	else
+		pim_show_interfaces(vrf_struct->info, vty, mlag_bool,
+				    json_parent);
+
+	if (uj)
+		vty_json(vty, json_parent);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_pim_interface_vrf_all,
+       show_ipv6_pim_interface_vrf_all_cmd,
+       "show ipv6 pim [mlag$mlag] vrf all interface [[detail|WORD]$detail] [json$json]",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       "MLAG\n"
+       VRF_CMD_HELP_STR
+       "PIM interface information\n"
+       "Detailed output\n"
+       "interface name\n"
+       JSON_STR)
+{
+	bool uj = !!json;
+	struct vrf *vrf_struct;
+	bool mlag_bool = !!mlag;
+	json_object *json_parent = NULL;
+	json_object *json_vrf = NULL;
+
+	if (uj)
+		json_parent = json_object_new_object();
+
+	RB_FOREACH (vrf_struct, vrf_name_head, &vrfs_by_name) {
+		if (!uj)
+			vty_out(vty, "VRF: %s\n", vrf_struct->name);
+		else
+			json_vrf = json_object_new_object();
+		if (detail)
+			pim_show_interfaces_single(
+				vrf_struct->info, vty, detail, mlag_bool, json_vrf);
+		else
+			pim_show_interfaces(vrf_struct->info, vty, mlag_bool,
+					    json_vrf);
+
+		if (uj)
+			json_object_object_add(json_parent, vrf_struct->name,
+					       json_vrf);
+	}
+	if (uj)
+		vty_json(vty, json_parent);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_pim_join,
+       show_ipv6_pim_join_cmd,
+       "show ipv6 pim [vrf NAME] join [X:X::X:X$s_or_g [X:X::X:X$g]] [json$json]",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       VRF_CMD_HELP_STR
+       "PIM interface join information\n"
+       "The Source or Group\n"
+       "The Group\n"
+       JSON_STR)
+{
+	pim_sgaddr sg = {0};
+	struct vrf *v;
+	struct pim_instance *pim;
+	json_object *json_parent = NULL;
+
+	v = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
+
+	if (!v) {
+		vty_out(vty, "%% Vrf specified: %s does not exist\n", vrf);
+		return CMD_WARNING;
+	}
+	pim = pim_get_pim_instance(v->vrf_id);
+
+	if (!pim) {
+		vty_out(vty, "%% Unable to find pim instance\n");
+		return CMD_WARNING;
+	}
+
+	if (!IPV6_ADDR_SAME(&s_or_g, &in6addr_any)) {
+		if (!IPV6_ADDR_SAME(&g, &in6addr_any)) {
+			memcpy(&sg.src, &s_or_g, sizeof(s_or_g));
+			memcpy(&sg.grp, &g, sizeof(g));
+		} else
+			memcpy(&sg.grp, &s_or_g, sizeof(s_or_g));
+	}
+
+	if (json)
+		json_parent = json_object_new_object();
+
+	pim_show_join(pim, vty, &sg, json_parent);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_pim_join_vrf_all,
+       show_ipv6_pim_join_vrf_all_cmd,
+       "show ipv6 pim vrf all join [json$json]",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       VRF_CMD_HELP_STR
+       "PIM interface join information\n"
+       JSON_STR)
+{
+	pim_sgaddr sg = {0};
+	struct vrf *vrf_struct;
+	json_object *json_parent = NULL;
+	json_object *json_vrf = NULL;
+
+	if (json)
+		json_parent = json_object_new_object();
+	RB_FOREACH (vrf_struct, vrf_name_head, &vrfs_by_name) {
```

Comment:
Add an extra line above this statement, it looks odd now.

Reply:
Done